model/infinispan/src/main/java/org/keycloak/models/cache/infinispan/RealmCacheSession.java

```diff
@@ -1187,16 +1227,15 @@ protected ClientModel cacheClient(RealmModel realm, ClientModel delegate, Long r
             cached = new CachedClient(revision, realm, delegate);
             adapter = new ClientAdapter(realm, cached, this);
 
-            long lifespan = model.getLifespan();
-            if (lifespan > 0) {
-                cache.addRevisioned(cached, startupRevision, lifespan);
-            } else {
-                cache.addRevisioned(cached, startupRevision);
-            }
+            long lifespan = model.getLifespan() < 0 ? cache.getConfiguredLifespan(model.getClass().getName()) : model.getLifespan();
+            long idleTime = cache.getConfiguredIdleTime(model.getClass().getName());
+            cache.addRevisioned(cached, startupRevision, lifespan, idleTime);
         } else {
             cached = new CachedClient(revision, realm, delegate);
             adapter = new ClientAdapter(realm, cached, this);
-            cache.addRevisioned(cached, startupRevision);
+            long lifespan = cache.getConfiguredLifespan(delegate.getClass().getName());
+            long idleTime = cache.getConfiguredIdleTime(delegate.getClass().getName());
+            cache.addRevisioned(cached, startupRevision, lifespan, idleTime);
```

Comment:
here is an example of using the config to set lifespan and idle time to stave off scaling memory concerns